fe/fe-core/src/main/java/com/starrocks/http/rest/StreamLoadMetaAction.java

```diff
@@ -0,0 +1,176 @@
+import io.netty.handler.codec.http.HttpMethod;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StreamLoadMetaAction extends RestBaseAction {
+
+    private static final Logger LOG = LogManager.getLogger(StreamLoadMetaAction.class);
+
+    public StreamLoadMetaAction(ActionController controller) {
+        super(controller);
+    }
+
+    public static void registerAction(ActionController controller) throws IllegalArgException {
+        controller.registerHandler(HttpMethod.GET,
+                "/api/{" + DB_KEY + "}/{" + TABLE_KEY + "}/_stream_load_meta",
+                new StreamLoadMetaAction(controller));
+    }
+
+    @Override
+    public void executeWithoutPassword(BaseRequest request, BaseResponse response) throws DdlException,
+            AccessDeniedException {
+        boolean enableBatchWrite = "true".equalsIgnoreCase(
+                request.getRequest().headers().get(StreamLoadHttpHeader.HTTP_ENABLE_BATCH_WRITE));
+        if (enableBatchWrite && redirectToLeader(request, response)) {
+            return;
+        }
+
+        String dbName = request.getSingleParameter(DB_KEY);
+        if (Strings.isNullOrEmpty(dbName)) {
+            StreamLoadMetaResult responseResult = new StreamLoadMetaResult(
+                    TStatusCode.INVALID_ARGUMENT.name(), ActionStatus.FAILED, "No database selected");
+            sendResult(request, response, responseResult);
+        }
+
+        String tableName = request.getSingleParameter(TABLE_KEY);
+        if (Strings.isNullOrEmpty(tableName)) {
+            StreamLoadMetaResult responseResult = new StreamLoadMetaResult(
+                    TStatusCode.INVALID_ARGUMENT.name(), ActionStatus.FAILED, "No table selected");
+            sendResult(request, response, responseResult);
+        }
+
+        Authorizer.checkTableAction(ConnectContext.get().getCurrentUserIdentity(), ConnectContext.get().getCurrentRoleIds(),
+                dbName, tableName, PrivilegeType.INSERT);
+
+        if (!enableBatchWrite) {
+            processNormalStreamLoad(request, response, dbName, tableName);
+        } else {
+            processBatchWriteStreamLoad(request, response, dbName, tableName);
+        }
+    }
+
+    private void processNormalStreamLoad(BaseRequest request, BaseResponse response, String dbName, String tableName) {
+        // currently there is no meta for normal stream load, and just return a success response.
+        StreamLoadMetaResult responseResult =
+                new StreamLoadMetaResult(TStatusCode.OK.name(), ActionStatus.OK, "");
+        sendResult(request, response, responseResult);
+    }
+
+    // batch write stream load =====================================================
+
+    // The parameter name to specify the type of metas.
+    private static final String TYPE_PARAM_NAME = "type";
```

Comment:
it's a little weird to have member variables located in the middle. May introduce a little trouble reading the code, those in-the-middle variables can be easily overlooked.

Reply:
Fixed